src/views/List.jsx

```diff
@@ -10,10 +30,31 @@ export function List({ data, listToken }) {
 			{data.length === 0 && (
 				<p>Your shopping list is empty. Click on "Add item" to begin!</p>
 			)}
-			<ul>
-				{data.map((item) => (
-					<ListItem key={item.id} name={item.name} />
-				))}
+			<form onSubmit={(e) => e.preventDefault()}>
+				<label htmlFor="filter">Filter items</label>
+				<br />
+				<input
+					type="text"
+					name="filter"
+					id="filter"
+					value={inputItem}
+					onChange={handleInput}
+					placeholder="Start typing here..."
+				/>
+				<Button
+					ariaLabel={'clear input field'}
+					label={'x'}
+					type={'reset'}
+					onClick={handleClear}
+				/>
+			</form>
+			{/* Accessibility feature: added the aria-live and tabIndex attributes for screenreaders to announce as the user types */}
+			<ul aria-live="polite">
+				{filterData.length > 0 ? (
+					filterData.map((item) => <ListItem key={item.id} name={item.name} />)
+				) : (
+					<p>No item found!</p>
+				)}
```

Comment:
(Giddy with excitement about this part! We made some changes on this page as well and I'm anticipating some conflicts! )

I agree with Raquel that on first render we could hide the "no result message," as well as when the list is empty, we could hide both the message and the filter.